bin/check_everything.py

```diff
@@ -0,0 +1,154 @@
+global_success &= CheckItemBits(SUBMITTY_INSTALL_DIR,True,"root","course_builders",0o751)
+global_success &= CheckItemBits(SUBMITTY_INSTALL_DIR+"/bin/untrusted_execute",False,"root","hwcron",0o550)
+
+
+# CHECK THE DATA DIRECTORY
+global_success &= CheckItemBits(SUBMITTY_DATA_DIR,True,"root","course_builders",0o751)
+global_success &= CheckItemBits(SUBMITTY_DATA_DIR+"/courses",True,"root","course_builders",0o751)
+
+
+# CHECK EACH COURSE
+for semester in os.listdir(SUBMITTY_DATA_DIR+"/courses"):
+    semester_path=SUBMITTY_DATA_DIR+"/courses/"+semester
+    if not os.path.isdir(semester_path):
+        continue
+    for course in os.listdir(semester_path):
+        course_path=semester_path+"/"+course
+        if not os.path.isdir(course_path):
+            continue
+        c_instructor=pwd.getpwuid(os.stat(course_path).st_uid).pw_name
+        c_group=grp.getgrgid(os.stat(course_path).st_gid).gr_name
+        global_success &= CheckCourseInstructorAndGroup(c_instructor,c_group)
+        print ("check course=", course, " inst=", c_instructor, " group=",c_group)
+        global_success &= CheckItemBits(course_path,True,c_instructor,c_group,0o770)
+
+        global_success &= CheckItemBits(course_path+"/build",True,c_instructor,c_group,0o770)
+        global_success &= CheckItemBits(course_path+"/ASSIGNMENTS.txt",False,c_instructor,c_group,0o660)
+
+        global_success &= CheckItemBits(course_path+"/config",True,c_instructor,c_group,0o770)
+        global_success &= CheckItemBits(course_path+"/config/config.ini",False,"hwphp",c_group,0o660)
+        global_success &= CheckItemBits(course_path+"/config/build",True,c_instructor,c_group,0o770)
+        global_success &= CheckItemBits(course_path+"/config/complete_config",True,c_instructor,c_group,0o770)
+        global_success &= CheckItemBits(course_path+"/config/form",True,c_instructor,c_group,0o770)
+
+        global_success &= CheckItemBits(course_path+"/bin",True,c_instructor,c_group,0o770)
+        global_success &= CheckItemBits(course_path+"/test_input",True,c_instructor,c_group,0o770)
+        global_success &= CheckItemBits(course_path+"/test_output",True,c_instructor,c_group,0o770)
+        global_success &= CheckItemBits(course_path+"/test_code",True,c_instructor,c_group,0o770)
+
+        global_success &= CheckItemBits(course_path+"/submissions",True,"hwphp",c_group,0o750)
+        global_success &= CheckItemBits(course_path+"/config_upload",True,"hwphp",c_group,0o750)
+        global_success &= CheckItemBits(course_path+"/results",True,"hwcron",c_group,0o750)
+        global_success &= CheckItemBits(course_path+"/checkout",True,"hwcron",c_group,0o750)
+
+        global_success &= CheckItemBits(course_path+"/reports",True,c_instructor,c_group,0o770)
+        global_success &= CheckItemBits(course_path+"/reports/summary_html",True,c_instructor,c_group,0o770)
+        global_success &= CheckItemBits(course_path+"/reports/all_grades",True,"hwphp",c_group,0o770)
```

Comment:
I'd almost want this to be a list of tuples that look like:
```
checks = [
    ("build", c_instructor, 0o770),
    ...
    ("submissions", "hwphp", 0o750),
    ...
]

for folder, user, mode in checks:
    global_success &= CheckItemBits(os.path.join(course_path, folder), True, user, c_group, mode)
```

which would make it a bit easier to add new checks/modify one, etc.